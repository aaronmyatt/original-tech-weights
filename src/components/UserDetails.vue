<template>
  <div class="card">
    <ul class="user-details list-group list-group-flush">
      <li class="user-name list-group-item">
        {{ userName }}
      </li>
      <li class="user-email list-group-item">
        {{ email }}
      </li>
      <li class="user-github list-group-item">
        {{ githubProfileUrl }}
      </li>
      <li class="user-linkedin list-group-item">
        {{ linkedinProfileUrl }}
      </li>
    </ul>
  </div>
</template>

<script>
export default {
  name: 'UserDetails',
  props: ['userName',
    'email',
    'githubProfileUrl',
    'linkedinProfileUrl',
  ],
};
</script>

<style scoped>
</style>
