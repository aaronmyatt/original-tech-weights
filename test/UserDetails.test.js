import { shallow } from 'vue-test-utils'
import UserDetails from '../src/components/UserDetails'

describe('UserDetails', () => {
  let cmp

  beforeEach(() => {
    cmp = shallow(UserDetails, {
        propsData: {
            userName: 'Test User',
            email: 'test@test.com',
            githubProfileUrl: 'github.com/test',
            linkedinProfileUrl: 'linkedin.com/test',
        }
    })
  })

  it('has the expected html structure', () => {
    expect(cmp.element).toMatchSnapshot()
  })

  describe('Props', () => {

      it('renders the users name', () => {
          expect(cmp.find('.user-details .user-name').text().trim()).toBe(cmp.vm.userName)
      })

      it('renders the users email', () => {
          expect(cmp.find('.user-details .user-email').text().trim()).toBe(cmp.vm.email)
      })

      it('renders the users github profile url', () => {
          expect(cmp.find('.user-details .user-github').html()).toContain(cmp.vm.githubProfileUrl)
      })

      it('renders the users linkdin url', () => {
          expect(cmp.find('.user-details .user-linkedin').html()).toContain(cmp.vm.linkedinProfileUrl)
      })

  })
})
