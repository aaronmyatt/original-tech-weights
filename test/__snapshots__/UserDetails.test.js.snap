// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UserDetails has the expected html structure 1`] = `
<div
  class="card"
>
  <ul
    class="user-details list-group list-group-flush"
  >
    <li
      class="user-name list-group-item"
    >
      
      Test User
    
    </li>
     
    <li
      class="user-email list-group-item"
    >
      
      test@test.com
    
    </li>
     
    <li
      class="user-github list-group-item"
    >
      
      github.com/test
    
    </li>
     
    <li
      class="user-linkedin list-group-item"
    >
      
      linkedin.com/test
    
    </li>
  </ul>
</div>
`;
